hyperon_das_atomdb/adapters/ram_only.py

```diff
@@ -73,40 +69,28 @@ def _add_atom_type(
             typedef_mark_hash, [name_hash, type_hash]
         )
 
-        try:
-            self.db.atom_type[key]
-        except KeyError:
+        atom_type = self.db.atom_type.get(key)
+        if atom_type is None:
             base_type_hash = ExpressionHasher.named_type_hash("Type")
             composite_type = [typedef_mark_hash, type_hash, base_type_hash]
             composite_type_hash = ExpressionHasher.composite_hash(
                 composite_type
             )
-
-            self.db.atom_type[key] = {
+            atom_type = {
                 '_id': key,
                 'composite_type_hash': composite_type_hash,
                 'named_type': _name,
                 'named_type_hash': name_hash,
             }
+            self.db.atom_type[key] = atom_type
+            self.named_type_table[name_hash] = _name
 
-        return self.db.atom_type[key]
-
-    def _add_names(self, _name: str, _type: str) -> Dict[str, str]:
-        key = self._node_handle(_type, _name)
-        try:
-            self.db.names[key]
-        except KeyError:
-            self.db.names[key] = _name
-        return self.db.names[key]
+        return atom_type
 
     def _add_outgoing_set(
         self, key: str, targets_hash: Dict[str, Any]
     ) -> None:
-        outgoing_set = self.db.outgoing_set.get(key)
```

Comment:
This condition if the key does not exist is not doing anything (it was an error due to the rush of the time) but, it is wrong, it was supposed to be an append, that is, `self.db.outgoing_set[key].append(targets_hash)`. But, I don't know if this condition would ever happen. Thinking about it, I don't think so, what do you think?

Reply:
It will not. I mean, for the `incoming_set` it does make sense to have the append but not for outgoing set. A link is something like Blah(X, Y, Z), a link Blah linking X, Y and Z. In this case, X, Y and Z are the outgoing set of a link of type Blah. This link will have a handle H calculated using its type and its outgoing set.

The outgoing set will map H -> (X, Y, Z).

If you add another link with the same handle H, it means it have the same type and the same targets, so it IS the same link. The map would remain the same: H -> (X, Y, Z). 

With `incoming_set` the history is different. This link above would add the following entries in inconing table:

X -> [H]
Y -> [H]
Z -> [H]

If you add another link that mentions X, for instance, it will be something like Blah(X, W) which will have a calculated handle H2. After you add it, we'll need to update incoming table like this:

X -> [H, H2]
Y -> [H]
Z -> [H]
W -> [H2]

I'm explaining this in detail because you need to understand the way information is represented/stored in DAS. Feel free to ask if it's still not clear.